docs/developer/extending/apps/synchronous-webhooks.mdx

```diff
@@ -0,0 +1,200 @@
+
+#### Listing all available payment gateways
+
+This query returns information about available payment gateways without any additional context:
+
+```graphql
+{
+  shop {
+    availablePaymentGateways(currency: "USD", channel: "default-channel") {
+      name
+      id
+      config {
+        field
+        value
+      }
+      currencies
+    }
+  }
+}
+```
+
+The `availablePaymentGateways` field accepts the optional `currency` argument; if provided, it will be passed in the webhook payload and can be used to return payment methods that support a particular currency.
+
+#### Listing available payment gateways for a checkout
+
+You can also query for available payment gateways in the context of a specific checkout. This is useful if payment methods depend on the current checkout (e.g. user’s shipping country etc.)
+
+```graphql
+{
+  checkout(token: "checkout-uuid-token") {
+    availablePaymentGateways {
+      name
+    }
+  }
+}
+```
+
+See the example payload sent from Saleor for this event: [List payment gateways payload](/developer/extending/apps/sample-webhook-payloads.mdx#list-payment-gateways-payload).
+
+#### Expected webhook response
+
+In the response, you should return a list of supported payment methods, for example:
+
+```json
+[
+   {
+      "id":"credit-card",
+      "name":"Credit Card",
+      "currencies":[
+         "EUR"
+      ],
+      "config":[]
+   },
+   {
+      "id":"apple-pay",
+      "name":"Apple Pay",
+      "currencies":[
+         "EUR",
+         "USD"
+      ],
+      "config":[
+         {
+            "field":"apple-pay-config-field",
+            "value":"value"
+         }
+      ]
+   }
+]
+```
+
+Each item on the list should have the following fields:
+- `id` - ID of a payment method; must be unique within your app.
+- `name` - the name of the payment method.
+- `currencies` - list of currencies supported by this payment method.
+- `config` - optional configuration, such as additional payment data that may be needed to proceed with the payment in the frontend.
+
+
+### Register webhooks for payment actions
+
+To subscribe to particular payment events (such as capture or authorization), you can have multiple webhooks with separate URLs for each action or a single webhook that handles all events (each webhook has an `X-Saleor-Event` header that contains the type of event).
+
+List of payment events: 
+
+- `PAYMENT_AUTHORIZE`
+- `PAYMENT_CAPTURE`
+- `PAYMENT_CONFIRM`
+- `PAYMENT_PROCESS`
+- `PAYMENT_REFUND`
+- `PAYMENT_VOID`
+
+:::note
+If an app has more than one webhook that listens to the same payment event, Saleor will always pick the first webhook from the list.
+:::
+
+See the example payload sent from Saleor for these payment events: [Payment actions payload](/developer/extending/apps/sample-webhook-payloads.mdx#payment-actions-payload).
+
+For payments that require confirmation (such as 3D Secure) Saleor triggers two events:
```

Comment:
`Saleor triggers two events:` - not exactly true.  Saleor will trigger payment_process for the first time. All next calls will trigger payment_confirm.  - it can be two or five calls, depends on how many actions the user needs to proceed.